core/file/path_spec.rb

```diff
@@ -3,6 +3,32 @@
 
 describe "File#path" do
   it_behaves_like :file_path, :path
+
+  ruby_version_is "2.5" do
+    platform_is :linux do
+      if defined?(File::TMPFILE)
```

Comment:
Can you use `guard -> { defined?(File::TMPFILE) } do` instead?
It allows to easily disable all guards which is useful in some cases (see https://github.com/ruby/mspec/issues/30#issuecomment-312487779 for details).